processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -76,56 +138,128 @@ public static ResponseContext createEmpty()
     return DefaultResponseContext.createEmpty();
   }
 
-  protected abstract Map<String, Object> getDelegate();
-
-  public Object put(String key, Object value)
+  public static ResponseContext deserialize(String responseContext, ObjectMapper objectMapper) throws IOException
   {
-    return getDelegate().put(key, value);
+    final Map<String, Object> delegate = objectMapper.readValue(
+        responseContext,
+        JacksonUtils.TYPE_REFERENCE_MAP_STRING_OBJECT
+    );
+    return new ResponseContext()
```

Comment:
Resulting ResponseContext depends on a TypeReference so in general in case of changing the TypeReference the resulting context should be also updated. To eliminate the possible resulting context update I used the inner class. If that fits I can add this description as a comment, if not I may remove inner class usage and use DefaultReponseContext as a resulting map.